lib/modules/dosomething/dosomething_signup/dosomething_signup.module

```diff
@@ -568,14 +568,22 @@ function dosomething_signup_get_mbp_params($account, $node, $opt_in) {
   );
   $campaign_link = dosomething_global_url('node/' . $nid, $url_options);
 
+  // Get node title, normal for collections, translatable for campaigns.
+  $title = '';
+  if (isset($node->title_field)) {
+    $title = $wrapper->language($language)->title_field->value();
+  } else {
+    $title = $node->title;
+  }
+
   $params = array(
     'email' => $account->mail,
     'uid' => $account->uid,
     'first_name' => dosomething_user_get_field('field_first_name', $account),
     'mobile' => dosomething_user_get_field('field_mobile', $account),
     'event_id' => $node->nid,
-    'campaign_title' => $wrapper->language($language)->title_field->value(),
-    'campaign_link' => dosomething_global_url('node/' . $node->nid, $url_options),
+    'campaign_title' => $title,
+    'campaign_link' => url('node/' . $node->nid, $url_options),
```

Comment:
I think this one should be `dosomething_global_url()`
See https://github.com/angaither/dosomething/commit/15f7281096e5ee0ef690a1c5de5511f87688dbf9
cc @blisteringherb